src/adapter/src/client.rs

```diff
@@ -385,157 +416,251 @@ impl SessionClient {
+                    match stmts.pop() {
+                        Some(stmt) => {
+                            // Mirror the behavior of the PostgreSQL simple query protocol.
+                            // See the pgwire::protocol::StateMachine::query method for details.
+                            self.start_transaction(Some(1)).await?;
                             results
-                                .push(SimpleResult::err("statement canceled due to user request"));
-                            continue;
+                                .push(self.execute_stmt(stmt, params.unwrap_or_default()).await?);
+                        }
+                        None => {
+                            if !params.unwrap_or_default().is_empty() {
+                                return Err(AdapterError::Unstructured(anyhow::anyhow!(
+                                    "cannot provide parameters to an empty query"
+                                )));
+                            }
                         }
-                    };
-                    let mut sql_rows: Vec<Vec<serde_json::Value>> = vec![];
-                    let col_names = match desc.relation_desc {
-                        Some(desc) => desc.iter_names().map(|name| name.to_string()).collect(),
-                        None => vec![],
-                    };
-                    let mut datum_vec = mz_repr::DatumVec::new();
-                    for row in rows {
-                        let datums = datum_vec.borrow_with(&row);
-                        sql_rows.push(datums.iter().map(From::from).collect());
                     }
-                    results.push(SimpleResult::Rows {
-                        rows: sql_rows,
-                        col_names,
-                    });
                 }
-                ExecuteResponse::Fetch { .. }
-                | ExecuteResponse::SetVariable { .. }
-                | ExecuteResponse::Tailing { .. }
-                | ExecuteResponse::CopyTo { .. }
-                | ExecuteResponse::CopyFrom { .. }
-                | ExecuteResponse::Raise { .. }
-                | ExecuteResponse::DeclaredCursor
-                | ExecuteResponse::ClosedCursor => {
-                    // NOTE(benesch): it is a bit scary to ignore the response
-                    // to these types of planning *after* they have been
-                    // planned, as they may have mutated state. On a quick
-                    // glance, though, ignoring the execute response in all of
-                    // the above situations seems safe enough, and it's a more
-                    // target allowlist than the code that was here before.
-                    results.push(SimpleResult::err(
-                        "executing statements of this type is unsupported via this API",
-                    ));
-                }
-            };
+            }
         }
-        Ok(SimpleExecuteResponse { results })
+        Ok(SqlHttpExecuteResponse { results })
```

Comment:
Need to commit the txn here.

Reply:
I'm hoping to sweep all of that up in https://github.com/MaterializeInc/materialize/pull/14582; this doesn't technically introduce a regression, and that PR is slated to get rebased on top of this with all of the tests to ensure the correct behavior--would you be cool with merging this as is and shifting that to the follow-on PR?

Reply:
I merged https://github.com/MaterializeInc/materialize/pull/14582 and rebased this on top of it, so this will be fixed on the next push